zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/metric/ZkClientPathMonitor.java

```diff
@@ -204,6 +224,22 @@ protected synchronized void record(int bytes, long latencyMilliSec, boolean isFa
     }
   }
 
+  /**
+   * Records metrics for async operations
+   */
+  protected synchronized void recordAsync(int bytes, long latencyMilliSec, boolean isFailure,
+      boolean isRead) {
+    if (isFailure) {
+      increaseAsyncFailureCounter(isRead);
+    } else {
+      increaseAsyncCounter(isRead);
+      increaseTotalLatency(isRead, latencyMilliSec);
```

Comment:
I suggest not to record the async latency into the total latency. It may skew the current latency data and cause confusion. We can choose not to record its latency or put into a different metric.